### PR TITLE
[SGP-13253] Make rate limiting on login case-insensitive

### PR DESCRIPTION
**Pull request by jcrumb**
The cache key needs to always be in a consistent case for rate limiting. Otherwise, it's possible to bypass the rate limiter by changing the casing of part or all of the email address passed to the login form.

### CODE REVIEW

File: allauth/account/tests.py [hunk 1]
```diff
@@ -536,6 +536,44 @@ def test_login_failed_attempts_exceeded(self):
...
+                {'login': 'ronald@example.com',
+                 'password': 'wrong'})
+            self.assertFormError(
+                resp,
+                'form',
+                None,
+                'Too many failed login attempts. Try again later.'
+                if is_locked
+                else
+                'The username and/or password you specified are not correct.')
```
Comment (jrfernandes) [inline allauth/account/tests.py:565]: No need to repeat all this, you can just call the test above here:
```suggestion
        self.test_login_failed_attempts_exceeded()
```

File: allauth/account/tests.py [hunk 2]
```diff
@@ -536,6 +536,44 @@ def test_login_failed_attempts_exceeded(self):
                 else
                 'The username and/or password you specified are not correct.')
 
+
```
Comment (jrfernandes) [inline allauth/account/tests.py:539]: Extra empty line here.

File: allauth/account/adapter.py [hunk 3]
```diff
@@ -464,7 +464,7 @@ def respond_email_verification_sent(self, request, user):
 
     def _get_login_attempts_cache_key(self, request, **credentials):
         site = get_current_site(request)
-        login = credentials.get('email', credentials.get('username', ''))
+        login = credentials.get('email', credentials.get('username', '')).upper()
```
Comment (jrfernandes) [inline allauth/account/adapter.py:467]: This should only be case-insensitive when using the email address for the key, not the username.

**Review by jrfernandes — COMMENTED** [review]
Nice, thank you for getting this done. I left some suggestions below.

**Comment by jcrumb** [comment]
Closing this, going to backport the exact upstream fix instead to minimize how much we're diverging. See [here](https://github.com/pennersr/django-allauth/commit/2fa7038a53e77dad504624584d2a533d08a3e7d1)

**Pull request closed without merging**